tests/test_permutation_test.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+from sklearn import datasets
+from omigami import permutation_test
+from omigami.feature_selector import FeatureSelector
+from omigami.utils import compute_t_student_p_value
+from omigami.data_structures import SelectedFeatures, ScoreCurve
+
+
+@pytest.fixture
+def fit_permutation_test():
+
+    fs = FeatureSelector(n_outer=5, metric="MISS", estimator="PLSC",)
+    pt = permutation_test.PermutationTest(feature_selector=fs, n_permutations=4)
+
+    pt.res = (
+        SelectedFeatures(min_feats=[0, 1], max_feats=[0, 1, 4, 3], mid_feats=[0, 1, 4]),
+        ScoreCurve(n_features=(2, 3, 4, 5), scores=(4.0, 4.5, 4.0, 4.75)),
+    )
+    pt.res_perm = [
+        (
+            SelectedFeatures(min_feats=[0], max_feats=[0, 1, 2], mid_feats=[0, 1]),
+            ScoreCurve(n_features=(1, 2, 3), scores=(4.0, 4.5, 4.5)),
+        ),
+        (
+            SelectedFeatures(min_feats=[0], max_feats=[0, 1, 2], mid_feats=[0, 1]),
+            ScoreCurve(n_features=(1, 2, 3), scores=(4.0, 4.5, 4.5)),
+        ),
+        (
+            SelectedFeatures(min_feats=[0], max_feats=[0, 1, 2], mid_feats=[0, 1]),
+            ScoreCurve(n_features=(1, 2, 3), scores=(4.0, 4.6, 4.9)),
+        ),
+        (
+            SelectedFeatures(min_feats=[0], max_feats=[0, 1, 2], mid_feats=[0, 1]),
+            ScoreCurve(n_features=(1, 2, 3), scores=(4.1, 4.6, 4.9)),
+        ),
+    ]
+    return pt
+
+
+@pytest.fixture
+def fit_feature_selector(results):
+    fs = FeatureSelector(n_outer=5, metric="MISS", estimator="PLSC",)
+    fs.is_fit = True
+    fs.results = results
+    sel_feats = fs.post_processor.select_features(results)
+    fs._selected_features = sel_feats
+    return fs
+
+
+def test_permutation_test():
+    fs = FeatureSelector(
+        n_outer=8, n_repetitions=8, random_state=0, estimator="PLSC", metric="MISS"
+    )
+    assert permutation_test.PermutationTest(feature_selector=fs, n_permutations=10)
+
+
+def test_fit(monkeypatch, fit_feature_selector):
```

Comment:
Are you using monkeypatch? It also seems unused further down.

Reply:
a leftover, thanks for catching it